components/eam/src/physics/crm/ndrop.F90

```diff
@@ -526,11 +524,13 @@ subroutine dropmixnuc( &
    end do
 
    call phys_getopts(use_MMF_out=use_MMF)
-   SPCAM_mmf    = (use_MMF .and. present(do_mmf)) ! SPCAM called from mmf section
-   SPCAM_notmmf = (use_MMF .and. .not. present(do_mmf)) ! SPCAM not called from mmf section
+
+   if ( (use_MMF .and. present(do_mmf)) .and. .not.present(species_class) ) then
+      call endrun('ERROR: species_class not provided for MMF call to dropmixnuc')
+   end if
 
 #if (defined MODAL_AERO)
-   if (SPCAM_mmf) then
+   if (use_MMF .and. present(do_mmf)) then
```

Comment:
What's the point of having both of these checks? Isn't the `do_mmf` flag sufficient?

Reply:
I think you're right for the new approach without the offline physics. In the old way "use_MMF" is always true, but "do_mmf" is not there when the offline physics is calling this. So I'm pretty sure that you're right that we can simplify this. I'll circle back to this and double check after I fix some other things.

Reply:
Changed this logic in recent commits. Decided to keep the "do_mmf" flag and omit the check for "use_MMF", but now "do_mmf" has to be present and true for the modifications to take effect.